docs/docs/user/references/components/overview.md

```diff
@@ -0,0 +1,32 @@
+# Overview 
+
+This section explains the different components of KPOps, 
+their usage and configuration in the pipeline 
+definition [`pipeline.yaml`](../../../resources/pipeline-components/pipeline).
+
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css">
+
+```mermaid
+flowchart BT
+    PipelineComponent --> BaseDefaultsComponent
+    KubernetesApp --> PipelineComponent
+    KafkaConnector --> PipelineComponent
+    KafkaApp --> KubernetesApp
+    StreamsApp --> KafkaApp
+    ProducerApp --> KafkaApp
+    KafkaSourceConnector --> KafkaConnector
+    KafkaSinkConnector --> KafkaConnector
+    
+    click KubernetesApp "../kubernetes-app"
+    click KafkaApp "../kafka-app"
+    click StreamsApp "../streams-app"
+    click ProducerApp "../producer-app"
+    click KafkaSourceConnector "../kafka-source-connector"
+    click KafkaSinkConnector "../kafka-sink-connector"
+```
+<p style="text-align: center;"><i>KPOps component hierarchy</i></p>
```

Comment:
I'd prefer to move this to [components_hierarchy.md](https://github.com/bakdata/kpops/pull/329/files#diff-834ea6d1f741693f9653559f64be6e2102babf370f839e25a8bc75e5e30d7a1e) and refer to it like so:

```
--8<--
./docs/resources/architecture/components-hierarchy.md
--8<--
```

Reply:
Done